webrtc/README.md

```diff
@@ -0,0 +1,268 @@
+by `/webrtc` and a multihash of the certificate that the node uses.
+
+Examples:
+- `/ip4/1.2.3.4/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`
+- `/ip6/fe80::1ff:fe23:4567:890a/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`
+
+## Connection Establishment
+
+### Browser to public Server
+
+Scenario: Browser _A_ wants to connect to server node _B_ where _B_ is publicly
+reachable but _B_ does not have a TLS certificate trusted by _A_.
+
+As a preparation browser _A_ [generates a
+certificate](https://www.w3.org/TR/webrtc/#dom-rtcpeerconnection-generatecertificate)
+and [gets the certificate's
+fingerprint](https://www.w3.org/TR/webrtc/#dom-rtccertificate-getfingerprints).
+
+1. Browser _A_ discovers server node _B_'s multiaddr, containing _B_'s IP, UDP
+  port, TLS certificate fingerprint and libp2p peer ID (e.g.
+  `/ip6/2001:db8::/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`),
+  through some external mechanism.
+
+2. _A_ instantiates a `RTCPeerConnection`, passing its local certificate as a
+   parameter. See
+   [`RTCPeerConnection()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection).
+
+3. _A_ constructs _B_'s SDP offer locally based on _B_'s multiaddr and sets it
+   via
+   [`RTCPeerConnection.setRemoteDescription()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setRemoteDescription).
+
+4. _A_ establishes the connection to _B_. _B_ learns of _A_ TLS fingerprint
+   through A's STUN message USERNAME attribute. See Open Questions below for
+   potential better solutions.
+
+5. See [Connection Security](#connection-security).
+
+6. See [Multiplexing](#multiplexing).
+
+#### Open Questions
+
+- How does the server learn the TLS certificate fingerprint of the browser? Is
+  embedding A's TLS certificate fingerprint in A's STUN message USERNAME
+  attribute the best option?
+
+- Is the fact that the server accepts STUN messages from the client prone to
+  attacks?
+
+    - Can an attacker launch an **amplification attack** with the STUN endpoint
+      of the server?
+
+    - Can a client run a **DOS attack** by sending many STUN messages with
+      different ufrags using different UDP source ports, forcing the server to
+      allocate a new peer connection for each? Would rate limiting suffice to
+      defend against this attack?
+
+  See also:
+  - https://datatracker.ietf.org/doc/html/rfc5389#section-16.2.1
+  - https://datatracker.ietf.org/doc/html/rfc5389#section-16.1.2
+
+- Do the major WebRTC server implementations support using the same UDP port for
+  multiple WebRTC connections, thus requiring multiplexing multiple WebRTC
+  connections on the same UDP port? In particular this came up for NodeJS.
+
+  This is related to [ICE Lite](https://www.rfc-editor.org/rfc/rfc5245), having
+  a host only advertise a single address, namely the host address, which is
+  assumed to be public.
+
+- Do the major (Go / Rust / ...) WebRTC implementations allow us to accept a
```

Comment:
[webrtc-rs](https://github.com/webrtc-rs/webrtc) will drop any connection attempts without having an ICE candidate pair (local-remote).

Reply:
Related discussion https://github.com/libp2p/specs/pull/412#discussion_r977888406.

Today rust-libp2p indexes by ufrag:

https://github.com/libp2p/rust-libp2p/pull/2622/files#diff-947bc76535b57b88959cee7a67a2ee0c8a3acf91ea796358c310a8b718782abeR69-R70

I will start another discussion on this matter. Resolving here for now.